include/xtypes/EnumeratedType.hpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2019, Proyectos y Sistemas de Mantenimiento SL (eProsima).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef EPROSIMA_ENUMERATED_TYPE_HPP_
+#define EPROSIMA_ENUMERATED_TYPE_HPP_
+
+#include <xtypes/PrimitiveType.hpp>
+
+#include <string>
+#include <map>
+#include <cassert>
+
+namespace eprosima {
+namespace xtypes {
+
+/// \brief DynamicType representing an enumerated type.
+/// An EnumeratedType represents a TypeKind::ENUMERATED_TYPE.
+template<typename T>
+class EnumeratedType : public PrimitiveType<T>
```

Comment:
Why inheritance instead of aggregation? Can this breaks the Ptr behaviour in some cases?

Reply:
Because we can see EnumeratedTypes as *restrictive* primitives. So, for most operations, they work the same.

I think it brokes the Ptr indeed, so I'm going to fix it.